### PR TITLE
Org name validation

### PR DESCRIPTION
**Pull request by sshmm**
### <samp>🤖 Generated by Copilot at 52e0994</samp>

This pull request adds a new feature to prevent name duplication among coordinape users and a new test case to verify its behavior. It modifies the `createOrgMembers` action handler to check for existing profile names before adding new org members, and returns an error if any name conflict is found. It also updates the `createOrgMembers.test.ts` file to include a scenario where two users have the same name but different addresses.

### <samp>🤖 Generated by Copilot at 52e0994</samp>

> _`createOrgMembers`_
> _checks names before adding them_
> _fall leaves no duplicates_

## Why

To reject duplicate names when adding new profiles for new org members

## Test and Deployment Plan

tested by createOrgMembers.test.ts

### <samp>🤖 Generated by Copilot at 52e0994</samp>

*  Prevent name duplication among coordinape users by checking existing profiles before creating org members ([link](https://github.com/coordinape/coordinape/pull/2138/files?diff=unified&w=0#diff-94334cc1c7f4caa25dc329463606e8404f55b378e1610eae5b7d5d8c20f78863R89-R123))
* Add a test case to verify the name conflict logic in the `createOrgMembers` action handler ([link](https://github.com/coordinape/coordinape/pull/2138/files?diff=unified&w=0#diff-9619bc278f9f361a18f4962a96937ae35c2eff2b3816183b552b6b5387a52893R250-R315))

### CODE REVIEW

File: api/hasura/actions/_handlers/createOrgMembers.ts [hunk 1]
```diff
@@ -86,6 +86,41 @@ async function handler(req: VercelRequest, res: VercelResponse) {
...
+  const { profiles: existingNames } = await adminClient.query(
+    {
+      profiles: [
+        {
+          where: {
+            _or: users.map(user => {
+              return {
+                _and: [
+                  { name: { _eq: user.name } },
+                  { address: { _nilike: user.address } },
```
Comment (crabsinger) [inline api/hasura/actions/_handlers/createOrgMembers.ts:99]: if the address DOES NOT match, but the name does, we still have an error though right? @sshmm

Comment (crabsinger) [reply api/hasura/actions/_handlers/createOrgMembers.ts:99]: should this just be to check if any profile name is already in use, regardless of address?

Comment (crabsinger) [reply api/hasura/actions/_handlers/createOrgMembers.ts:99]: nevermind - this is correct.

**Review by crabsinger — APPROVED** [review]
nice work on this @sshmm !

File: api/hasura/actions/_handlers/createOrgMembers.ts (hunk 1 above)
Comment (sshmm) [reply api/hasura/actions/_handlers/createOrgMembers.ts:99]: May be that user already has the a profile for this address so if the name and address matches the profile it won't throw an error.
What it does is that it fetches  all profiles with names from the input as long as these names are assigned for profiles with different addresses from the input addresses.

**Pull request merged**